src/ripple/beast/insight/impl/Metric.cpp

```diff
@@ -27,10 +25,6 @@
 namespace beast {
 namespace insight {
 
-Base::~Base() = default;
-
-BaseImpl::~BaseImpl() = default;
-
 CounterImpl::~CounterImpl() = default;
```

Comment:
Since you're here, why not just "inline" these in the relevant classes? e.g. just `~CounterImpl() = default;` in the header.

Reply:
It's not a huge deal, but I am not a fan of inlining things that the compiler has to outline anyway.  Virtual functions are a case in point.  If you inline it, the compiler is going to outline it in every translation unit it appears.  And then the linker  is going to unique all of those outlined copies.

Additionally the Itanium ABI has this concept of _key_ _function_ (https://itanium-cxx-abi.github.io/cxx-abi/abi.html#vague-vtable).  The _key_ _function_ is the first non-pure virtual function that is not inline at the point of class definition.  The TU where the _key_ _function_ is where the virtual table and RTTI information will also be outlined.

If the _key_ _function_ does _not_ exist, then the v-table and the RTTI information is emitted in each TU, and the linker has to unique them. 

This really adds up.  So my guideline is to not inline virtual functions.